engine/include/cubos/engine/collisions/contact_manifold.hpp

```diff
@@ -15,6 +15,92 @@
 
 namespace cubos::engine
 {
+    /// @brief Indicates the type of a geometric feature (vertex, edge, or face) and the feature index. Currently each
+    /// feature index is simply the index in the array with that information in the shape.
+    class FeatureId
```

Comment:
Since this is in the `cubos::engine` namespace, you should give this a more specific name, e.g. `ManifoldFeatureId`. You can also make it a type internal to other existing types.